src/HealthChecks.Nats/NatsHealthCheck.cs

```diff
@@ -0,0 +1,100 @@
+﻿namespace HealthChecks.Nats
+{
+    using Microsoft.Extensions.Diagnostics.HealthChecks;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using NATS.Client;
+    using System.Collections.Generic;
+    using System;
+
+    /// <summary>
+    /// Health check for Nats Server
+    /// </summary>
+    public class NatsHealthCheck : IHealthCheck
+    {
+        /// <summary>
+        ///  Provides factory methods to create connections to NATS Servers.
+        /// </summary>
+        private readonly ConnectionFactory _connectionFactory = new ConnectionFactory();
+        
+        private readonly NatsOptions _options;
+
+        /// <summary>
+        /// An NATS.Client.IConnection object connected to the NATS server.
+        /// </summary>
+        private IConnection _connection = null;
```

Comment:
Created connection implements `IDisposable`: https://github.com/nats-io/nats.net/blob/master/src/NATS.Client/Conn.cs#L101 so you should to implement `IDisposable` on `NatsHealthCheck`.

Reply:
Commit: b1e6ad14cadedf1eaa0e0d21722551d351167da6 [b1e6ad1]
Parents: c839f8fa6a
PR 891 disposable, editor config